web/src/app/users/UserContactMethodVerificationForm.js

```diff
@@ -0,0 +1,95 @@
+import React, { useEffect } from 'react'
+import p from 'prop-types'
+import Grid from '@material-ui/core/Grid'
+import TextField from '@material-ui/core/TextField'
+import LoadingButton from '../loading/components/LoadingButton'
+import gql from 'graphql-tag'
+import { makeStyles } from '@material-ui/core/styles'
+import { FormContainer, FormField } from '../forms'
+import { useMutation } from '@apollo/react-hooks'
+
+/*
+ * Triggers sending a verification code to the specified cm
+ * when the dialog is first opened
+ */
+const sendVerificationCodeMutation = gql`
+  mutation sendContactMethodVerification(
+    $input: SendContactMethodVerificationInput!
+  ) {
+    sendContactMethodVerification(input: $input)
+  }
+`
+
+const useStyles = makeStyles({
+  fieldGridItem: {
+    flexGrow: 1,
+  },
+  sendGridItem: {
+    display: 'flex',
+    alignItems: 'center',
+  },
+})
+
+export default function UserContactMethodVerificationForm(props) {
+  const classes = useStyles()
+
+  const [sendCode] = useMutation(sendVerificationCodeMutation, {
+    variables: {
+      input: {
+        contactMethodID: props.contactMethodID,
+      },
+    },
+  })
+
+  // componentDidMount
+  useEffect(() => {
+    sendCode().catch(err => props.setSendError(err.message))
+  }, [])
+
+  return (
+    <FormContainer optionalLabels {...props}>
+      <Grid container spacing={2}>
+        <Grid item className={classes.sendGridItem}>
+          <LoadingButton
+            color='primary'
+            loading={status.loading}
+            disabled={props.disabled}
+            buttonText={'Resend Code'}
+            noSubmit
+            onClick={() =>
+              sendCode().catch(err => props.setSendError(err.message))
```

Comment:
Should we just define this in a const and use it here and above instead of duplicating?

```js
const doSendCode = () => sendCode.catch( ... )

useEffect(doSendCode, [])
// ...
onClick={doSendCode}
```